subprojects/core/src/main/java/org/gradle/process/internal/ExecFactory.java

```diff
@@ -29,10 +30,10 @@
     /**
      * Creates a new factory for the given context.
      */
-    ExecFactory forContext(FileResolver fileResolver, FileCollectionFactory fileCollectionFactory, Instantiator instantiator);
+    ExecFactory forContext(FileResolver fileResolver, FileCollectionFactory fileCollectionFactory, Instantiator instantiator, ObjectFactory objectFactory);
```

Comment:
We should think about cleaning the debt introduced by the migration from `Instantiator` to `ObjectFactory`. In the end, both of them are the same thing, where `ObjectFactory` is the public API to `Instantiator`. I added this topic to the cross-pollination day.